packages/devtools_app/lib/src/screens/memory/framework/memory_tabs.dart

```diff
@@ -42,42 +43,49 @@ class MemoryTabView extends StatelessWidget {
     );
   }
 
-  List<({DevToolsTab tab, Widget tabView})> _generateTabRecords() {
-    return [
-      (
+  TabAndView _diff() => (
         tab: DevToolsTab.create(
-          key: MemoryScreenKeys.dartHeapTableProfileTab,
-          tabName: 'Profile Memory',
+          key: MemoryScreenKeys.diffTab,
           gaPrefix: _gaPrefix,
+          tabName: 'Diff Snapshots',
         ),
         tabView: KeepAliveWrapper(
-          child: AllocationProfileTableView(
-            controller: controller.profile,
+          child: DiffPane(
+            diffController: controller.diff,
           ),
         ),
-      ),
-      (
+      );
+
+  TabAndView _profile() => (
         tab: DevToolsTab.create(
-          key: MemoryScreenKeys.diffTab,
+          key: MemoryScreenKeys.dartHeapTableProfileTab,
+          tabName: 'Profile Memory',
           gaPrefix: _gaPrefix,
-          tabName: 'Diff Snapshots',
         ),
         tabView: KeepAliveWrapper(
-          child: DiffPane(
-            diffController: controller.diff,
+          child: AllocationProfileTableView(
+            controller: controller.profile!,
           ),
         ),
-      ),
-      (
+      );
+
+  TabAndView _trace() => (
         tab: DevToolsTab.create(
           key: MemoryScreenKeys.dartHeapAllocationTracingTab,
           tabName: 'Trace Instances',
           gaPrefix: _gaPrefix,
         ),
         tabView: KeepAliveWrapper(
-          child: TracingPane(controller: controller.tracing),
+          child: TracingPane(controller: controller.tracing!),
         ),
-      ),
+      );
+
+  List<TabAndView> _generateTabRecords() {
+    final hasData = controller.mode != ControllerCreationMode.disconnected;
+    return [
+      if (hasData) _profile(),
+      _diff(),
```

Comment:
can you add a comment her explaining why we are still showing the diff view if there is no data? I assume this is because you can load an individual snapshot into the diff pane?